pyproject.toml

```diff
@@ -26,10 +27,9 @@ classifiers = [
    "Programming Language :: Python :: 3.10",
    "Topic :: Software Development :: Libraries",
 ]
-requires-python = ">= 3.7, < 4"
+requires-python = ">= 3.11, < 4"
 dependencies = [
-    "googleapis-common-protos >= 1.56.2, < 2",
-    "grpcio >= 1.47.0, < 2",
```

Comment:
This should be kept here, we need it in the python users to make sure the dependencies are fulfilled. Our (generated) code import stuff from the `googleapis-common-protos` python package.

```console
$ rg import.*google .direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/ 
.direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/microgrid_pb2_grpc.pyi
15:import google.protobuf.empty_pb2
16:import google.protobuf.wrappers_pb2

.direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/microgrid_pb2.pyi
20:import google.protobuf.descriptor
21:import google.protobuf.internal.containers
```

Otherwise users will need to declare these dependencies themselves in their projects even if they don't use the google stuff in their code.

We also need to leave `grpcio`, stuff from the package `grpc` (provided by `grpcio` is used in the generated code:

```console
$ rg grpc .direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/
.direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/common_pb2_grpc.py
3:import grpc

.direnv/python-3.11.2/lib64/python3.11/site-packages/frequenz/api/microgrid/microgrid_pb2_grpc.pyi
17:import grpc
20:    def __init__(self, channel: grpc.Channel) -> None: ...
...
```

Reply:
Reverted :rocket: